cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/app/etl/realtime/ETLWorker.java

```diff
@@ -212,7 +213,9 @@ private void initializeSinks(WorkerContext context) throws Exception {
         transform.initialize(transformContext);
         StageMetrics stageMetrics = new StageMetrics(metrics, PluginID.from(transformId));
         transformDetailList.add(new TransformDetail(transformId, transform, stageMetrics));
-        tranformIdToDatasetName.put(transformId, transformIds.get(i).getErrorDatasetName());
+        if (transformIds.get(i).getErrorDatasetName() != null) {
```

Comment:
this collection is a misleading name, should be transformInfos instead of transformIds